### PR TITLE
Use caching to speed up CI on GH workflow

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by othercorey** [comment]
I don't know how to review this or know if it works, but I see this output for example in `PHP 7.2 & sqlite`

> Cache not found for input keys: Linux-composer-9acc5a587e3793a4a11b986d4f0f8b53e6550c393541eefd60ecdd48c2d26ed3, Linux-composer-.

> [warning]Cache already exists. Scope: refs/pull/14322/merge, Key: Linux-composer-9acc5a587e3793a4a11b986d4f0f8b53e6550c393541eefd60ecdd48c2d26ed3, Version: (null)

**Comment by garas** [comment]
`Cache not found for input keys` - I think it is normal on first run without no cache existing. Next run should be OK.

`[warning]Cache already exists` - _Coding Standard_ job finished first and has no warning and successfully saved cache file, other jobs finished later and tried to upload with same cache key.

**Comment by othercorey** [comment]
I tried to re-run the tests to see what happened now that the cache was initialized, but github actions doesn't let you do this unless there is an error?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -102,6 +112,16 @@ jobs:
...
         sqlcmd -S "(localdb)\MSSQLLocalDB" -Q "create database cakephp;"
 
+    - name: Get composer cache directory
+      id: composer-cache
+      run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+    - name: Cache composer dependencies
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
```
Comment (markstory) [inline .github/workflows/ci.yml]: Our composer.json rarely changes. Wouldn't this prevent us from having tests run against new patch releases of our dependencies as they are released?

Comment (ADmad) [reply .github/workflows/ci.yml]: Yeah ideally the `composer.lock` file should be used, but we don't have that committed in the repo.

Comment (ADmad) [reply .github/workflows/ci.yml]: > https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

Unused cache keys are removed after 7 days but I don't see any info about forcibly clearing cache. So we can just take a break of 7 days every month or so to clear the cache :smile:.

Comment (markstory) [reply .github/workflows/ci.yml]: Won't our key always be 'used'? Could we include a date based component to the cache key so it automatically rotates once a week?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,6 +44,18 @@ jobs:
...
         coverage: pcov
 
+    - name: Get composer cache directory
+      id: composer-cache
+      run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+    - name: Cache composer dependencies
+      uses: actions/cache@v1
+      env:
+        KEY_DATE=$(date '+%y-%m')
```
Comment (markstory) [inline .github/workflows/ci.yml]: Why is this one different than the others?

Comment (ADmad) [reply .github/workflows/ci.yml]: It's still WIP. I still haven't figured how to get the required date string is a variable which I can use in the next step :)

**Comment by ADmad** [comment]
@markstory The cache key now contains a date component so that that key gets changed every month.

**Pull request merged**